flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/UnionResultSubpartitionView.java

```diff
@@ -71,18 +73,35 @@ public class UnionResultSubpartitionView
     private final Queue<Tuple2<ResultSubpartition.BufferAndBacklog, Integer>> cachedBuffers =
             new LinkedList<>();
 
+    /**
```

Comment:
Though `cachedBuffers` is not introduced by this PR, l also left a comment here. They should be recycled whenever an exception occurs.
And when `releaseAllResources`, I think all the `cachedBuffers` should also be recycled in case of leaking buffers.

And when `releaseAllResources`, I am not sure it is necessary to clear these collections (e.g., `unregisteredAvailableViews`) as well.